diff --git a/src/desktop/desktop.c.orig b/src/desktop/desktop.c
index 2a15012..c4ea230 100644
--- a/src/desktop/desktop.c.orig
+++ b/src/desktop/desktop.c
@@ -27,6 +27,7 @@
 #ifdef DESKTOP_INTEGRATION
 
 #include <gtk/gtk.h>
+#include <gdk/gdkx.h>
 //#include "fm-desktop.h"
 
 #include "vfs-file-info.h"
@@ -73,6 +74,12 @@ void fm_turn_on_desktop_icons(gboolean transparent) {
     gint i;
     int big = 0;
 
+    gdpy = gdk_display_get_default();
+    #if GTK_CHECK_VERSION (3, 0, 0)
+        if( ! GDK_IS_X11_DISPLAY( gdpy ) )
+            return;
+    #endif
+
     if( ! group )
         group = gtk_window_group_new();
 
@@ -80,8 +87,6 @@ void fm_turn_on_desktop_icons(gboolean transparent) {
 
     vfs_mime_type_get_icon_size( &big, NULL );
 
-    gdpy = gdk_display_get_default();
-
     n_screens = gdk_display_get_n_screens( gdpy );
     desktops = g_new( GtkWidget *, n_screens );
     for ( i = 0; i < n_screens; i++ )
diff --git a/src/main-window.c.orig b/src/main-window.c
index d49b63a..7430d84 100644
--- a/src/main-window.c.orig
+++ b/src/main-window.c
@@ -3263,7 +3263,11 @@ static long get_desktop_index( GtkWindow* win ) {
     } else {
         // get current desktop
         display = gdk_display_get_default();
+#if GTK_CHECK_VERSION (3, 0, 0)
+       if ( display && GDK_IS_X11_DISPLAY (display ) )
+#else
         if ( display )
+#endif
             window = gdk_x11_window_lookup_for_display( display, gdk_x11_get_default_root_xwindow() );
     }
 
diff --git a/src/main.c.orig b/src/main.c
index 63a24f0..ae34b82 100644
--- a/src/main.c.orig
+++ b/src/main.c
@@ -311,12 +311,23 @@ gboolean on_socket_event( GIOChannel* ioc, GIOCondition cond, gpointer data ) {
 }
 
 void get_socket_name_nogdk( char* buf, int len ) {
-    char* dpy = g_strdup( g_getenv( "DISPLAY" ) );
-    if ( dpy && !strcmp( dpy, ":0.0" ) )
+    char* dpy;
+#if GTK_CHECK_VERSION(3, 0, 0)
+   const char* tmp = g_getenv( "WAYLAND_DISPLAY" );
+   if ( tmp )
+       dpy = g_strdup( tmp );
+   else
+#else
+   if ( TRUE )
+#endif
     {
-        // treat :0.0 as :0 to prevent multiple instances on screen 0
-        g_free( dpy );
-        dpy = g_strdup( ":0" );
+       dpy = g_strdup( g_getenv( "DISPLAY" ) );
+       if ( dpy && !strcmp( dpy, ":0.0" ) )
+       {
+           // treat :0.0 as :0 to prevent multiple instances on screen 0
+           g_free( dpy );
+           dpy = g_strdup( ":0" );
+       }
     }
     g_snprintf( buf, len, "%s/.zzzfm-socket%s-%s", xset_get_tmp_dir(), dpy, g_get_user_name() );
     g_free( dpy );
diff --git a/src/vfs/vfs-execute.c.orig b/src/vfs/vfs-execute.c
index d2eb4fa..af710f7 100644
--- a/src/vfs/vfs-execute.c.orig
+++ b/src/vfs/vfs-execute.c
@@ -6,6 +6,8 @@
 
 #include "vfs-execute.h"
 
+#include <gtk/gtk.h>
+#include <gdk/gdkx.h>
 #include <string.h>
 #include <stdlib.h>
 
diff --git a/src/desktop/working-area.c.orig b/src/desktop/working-area.c
index bcd756c..d77c2eb 100644
--- a/src/desktop/working-area.c.orig
+++ b/src/desktop/working-area.c
@@ -17,6 +17,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
+# include <gtk/gtk.h>
 # include <gdk/gdk.h>
 # include <gdk/gdkx.h>
 # include <X11/Xlib.h>
@@ -37,7 +38,11 @@ gboolean  gf_display_get_workarea(GdkScreen* g_screen, GdkRectangle *rect) {
 
 	/* get the gdk display */
 	g_display = gdk_display_get_default();
+#if GTK_CHECK_VERSION(3, 0, 0)
+       if(!g_display || !GDK_IS_X11_DISPLAY(g_display))
+#else
 	if(!g_display)
+#endif
 		return FALSE;
 
 	/* get the x display from the gdk display */
